audmath/core/api.py

```diff
@@ -193,6 +194,86 @@ def inverse_normal_distribution(
     return _force_float(x)
 
 
+def rms(
+        x: typing.Union[float, typing.Sequence[float], np.ndarray],
+) -> float:
+    r"""Root mean square.
+
+    The root mean square
+    for a signal of length :math:`N`
+    is given by
+
+    .. math::
+
+        \sqrt{\frac{1}{N} \sum_{n=1}^N x_n^2}
+
+    where :math:`x_n` is the value
+    of a single sample
+    of the signal.
+
+    For an empty signal
+    :const:`np.NaN` is returned.
+
+    Args:
+        x: input signal
+
+    Returns:
+        root mean square of input signal
+
+    Example:
+        >>> rms([0, 1])
+        0.7071067811865476
+
+    """
+    # Don't raise warning for empty signal
+    # when returning NaN
+    with warnings.catch_warnings():
```

Comment:
We could also handle the empty signal case separately, might be easier to understand and we do not rely on the behavior of `numpy`.

Reply:
I was afraid that it would be slower and I also wasn't sure what is the best way to catch those cases. `len(x)` or `len(np.array(x))` will not work as you can also provide a single value to the function.

Reply:
But maybe `catch_warnings` is also slow?

Reply:
Maybe: `if not isinstance(x, float)  and len(x) == 0`?

Reply:
It does still not cover all cases as:

```python
>>> len(np.array([[]]))
1
```

;)

So I guess we have to convert everything to a `numpy` array and then check for size.

Reply:
Maybe we should first discuss if it really makes sense to support scalars?

Reply:
Otherwise we might also rely on `audeer.to_list()` here.

Reply:
A simple solution would be to use `x.size` instead of `len(x)`, then the following should work:

```python
x = np.array(x)
if x.size == 0:
    ...
else:
    ...
```

Reply:
Hehe, just saw that you proposed that already :)